Automattic/sensei
# Add/lm template settings api

Fixes #5537 

### Changes proposed in this Pull Request

* Adds the "Appearance" tab into the Sensei Settings.
* Moves the Learning Mode settings into "Appearance" tab.
* Adds the "Learning Mode Templates" settings that allows user to choose the desired block template for the Learning Mode. **Note**: This is a simplified version of the templates setting for the time being. The fancy UI version will be implemented at #5539 .
* Adds the ability to add custom, third-party block templates for the Learning Mode.

### Testing instructions


* Go to Sensei Settings and confirm there is a new Appearance tab available.
* In the appearance tab, confirm there is a "Learning Mode Templates" setting that lists the templates user can choose from. Confirm that the premium templates are disabled and has upsells instead. Confirm the Default template is selected by default. 
* Switch to `add/lm-templates` branch on the Sensei Pro and build the assets.
* Refresh the Sensei Settings page and confirm that the premium templates are now can be selected and the upsells are gone.
* Select one of the premium templates and save the settings.
* Go to "WP Admin -> Appearance -> Editor (beta)". Confirm the Learning Mode templates are listed on the top.
* Click on the Lesson ( Learning Mode - ...) template and confirm the premium template is being used. At the moment premium templates are the same as the Default template except it has different colors and different number of **Notices** blocks in them. (Modern 🟩 , Video 🟦 , Video Full 🟨 ).
* Confirm the new premium template is also applied on the frontend. Visit a Lesson page with Learning Mode enabled.
* Edit the template. Move the blocks around and change some colors. Save. Confirm the updates reflect on the frontend.
* Try with block theme and non-block themes.
* When sensei-pro is uninstalled( or simply switch back to `trunk`), the Learning Mode should keep working simply falling back to the Default template.


### New/Updated Hooks

#### [sensei_learning_mode_block_templates](https://github.com/Automattic/sensei/blob/1417682d25dc005021cc3bddad14f6ea1f8b7b6e/includes/course-theme/class-sensei-course-theme-template-selection.php#L114-L125)

```php”
/**
 * Filters the Learning Mode block templates list. Allows to add additional ones too.
 *
 * @since $$next-version$$
 * @hook  sensei_learning_mode_block_templates
 *
 * @param Sensei_Course_Theme_Template[] $templates {
 *     The list of Learning Mode block templates. If adding a new template then it's key
 *     should be the template name.
 *
 * @return Sensei_Course_Theme_Template[] The list of extra learning mode block templates.
 */
```
